internal/sqlsmith: TestGenerateParse failed

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2875569&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2875569&tab=artifacts#/) on master @ [8a2143878c560466294742cf81d2ab3f79797915](https://github.com/cockroachdb/cockroach/commits/8a2143878c560466294742cf81d2ab3f79797915):


```
								(
									'01030000400100000004000000BC56BD6162EAF6C1189C2C7C244502C2DEE9EF8E5151FBC16E40A8BA8AA9F64180DC26CBC70EDA41F080E4777626D741580BFBDA55C6D241C023C2D06A9CD041E6D7C971222F0142BC56BD6162EAF6C1189C2C7C244502C2DEE9EF8E5151FBC1':::GEOMETRY,
								),
								(-1385707238):::INT8
							),
							(
								(
									'0103000080010000000A00000078FDBB33F90202C2FC5D90ED0F97E241E1C1BA8CEA35F7C1F80BA5259CFEFDC1186BCDE31CADD04134AEA8BBEE99EA41A0B8819E6835B1C1FCB1C2BF77B0E6C16A13051AF268F041645958994115F0418704810070D6F4C1FC5CE332775FFCC16845DD1E32DCDD41208B137C4ED7B0C1D0F06B82542DE941FA5533853D45F94170F38FCF3CADF741AC306E9767CEEBC13890846E2400EB410C2B8C4B6E95FC41A09F62945037B541883CB0E5FCB2ED41105710159E080242002B854E94E482C11C120C0E0E55F5C1B20258415591F941F05BE53AF616E44178FDBB33F90202C2FC5D90ED0F97E241E1C1BA8CEA35F7C1':::GEOMETRY,
								),
								1890654611:::INT8
							)
					)
						AS tab_73 (col_140, col_141)
		)
SELECT
	'{"M+mqt]aADW": "&W?EpRB", "a": {"4,g{<>f ": 0.9381419049035633, "c": {"bar": null, "baz": {"a": "}dm``}NHdlJg", "b": {"W+S=/^<": {}, "^:*^": false}}}, "foo": 0.33594010998333496, "k@|}/W(~CaG[": {"baz": []}}}':::JSONB
		AS col_142,
	tab_74.col1_1 AS col_143,
	e':G0\x10\r\tZ':::STRING AS col_144,
	71410660:::OID AS col_145
FROM
	defaultdb.public.table1@[0] AS tab_74, with_28 AS cte_ref_7
LIMIT
	58:::INT8;

STMT: 47
ROLLBACK TRANSACTION;

STMT: 48
WITH
	with_29 (col_146) AS (SELECT * FROM (VALUES ((-22341):::INT8)) AS tab_75 (col_146))
SELECT
	cte_ref_8.col_146 AS col_147
FROM
	with_29 AS cte_ref_8
GROUP BY
	cte_ref_8.col_146
ORDER BY
	cte_ref_8.col_146 DESC, cte_ref_8.col_146 DESC
LIMIT
	41:::INT8;

STMT: 49
SELECT count(*) AS col_148 FROM defaultdb.public.table2@[0] AS tab_76 GROUP BY tab_76.col2_9 LIMIT 74:::INT8;

    sqlsmith_test.go:190: INSERT INTO defaultdb.public.table1 AS tab_77 VALUES (): at or near ")": syntax error
W210413 05:53:53.623076 10871 jobs/registry.go:729  [-] 645  canceling all adopted jobs due to stopper quiescing
W210413 05:53:53.623058 10870 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 644  exiting heartbeat loop
I210413 05:53:53.644540 10814 kv/kvclient/kvcoord/transport_race.go:106  [-] 646  transport race promotion: ran 38 iterations on up to 844 requests
--- FAIL: TestGenerateParse (2.92s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestGenerateParse PKG=./pkg/internal/sqlsmith TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/internal/sqlsmith/relational.go
+++ b/pkg/internal/sqlsmith/relational.go
@@ -972,6 +972,11 @@ func (s *Smither) makeInsert(refs colRefs) (*tree.Insert, *tableRef, bool) {
 				row = append(row, rowenc.RandDatum(s.rnd, tree.MustBeStaticallyKnownType(c.Type), !notNull))
 			}
 		}
+
+		if len(names) == 0 {
+			return nil, nil, false
+		}
+
 		insert.Columns = names
 		insert.Rows = &tree.Select{
 			Select: &tree.ValuesClause{

